xmake/modules/utils/archive/extract.lua

```diff
@@ -33,7 +33,7 @@ import("extension", {alias = "get_archive_extension"})
 function _extract_using_tar(archivefile, outputdir, extension, opt)
 
     -- the tar of windows can only extract "*.tar"
-    if os.host() == "windows" and extension ~= ".tar" then
+    if os.host() == "windows" and not os.is_subhost("msys", "cygwin") and extension ~= ".tar" then
```

Comment:
no, msys2/cygwin's tar does not decompress many non-.tar files. So I disabled it as well. Had a lot of problems with it before

Reply:
Ok. I'll remove this